prow/config/jobs_test.go

```diff
@@ -76,8 +79,8 @@ func TestPresubmits(t *testing.T) {
 			}
 			// Check that the merge bot will run AlwaysRun jobs, otherwise it
 			// will attempt to rerun forever.
-			if job.AlwaysRun && !job.re.MatchString(testThis) {
-				t.Errorf("AlwaysRun job %s: \"%s\" does not match regex \"%v\".", job.Name, testThis, job.Trigger)
+			if job.AlwaysRun && !job.re.MatchString(newTestThis) && !job.re.MatchString(oldTestThis) {
```

Comment:
Every job with `AlwaysRun` must match **all** of the following:
* `/test all EXTRA STUFF`
* `@k8s-bot test this EXTRA STUFF`
* `/test all`
* `@k8s-bot test this`

not just one of them

Also, if `job.AlwaysRun == false` we should validate it does not match any of them (aka `job.AlwaysRun == job.re.MatchString(newTestThisAndTheOtherOnes)`).